### ISSUE TITLE
Best practice for using RENV_PATHS_LOCAL with collaborators

### ISSUE THREAD

**Issue by dchiu911** [rstudio/renv#550]
I've gone through https://rstudio.github.io/renv/articles/local-sources.html and have been using the `${RENV_PATHS_LOCAL}/<package>_<version>.tar.gz` syntax in our various project `.Renviron`. My question is what the best practice should be when working with collaborators:
- Agree on a common `RENV_PATHS_LOCAL` on each of our machines where we put the local source packages
- Have a separate `RENV_PATHS_LOCAL` for each user and educate the workflow to specify in their own `.Renviron` (and to gitignore this so it won't overwrite others')

I should mention that we have local packages that are used across different projects.

**Comment by kevinushey** [comment]
There isn't really a best practice here since it's not yet commonly done. In the ideal world, I think you'd have some sort of shared drive (network file system?) that you could point to.

Although, at that point, it seems like you might as well consider setting up your own local R package repository, with something like RSPM / drat / miniCRAN.

**Comment by dchiu911** [comment]
We do have a shared network drive where we use as sort of a local repo. But our current workflow is to copy the source package from the network drive and put it in `RENV_PATHS_LOCAL=~/some/path/to/package/repo` on our local machines. Perhaps I could change the path to point to the network drive but then I don't think Windows and macOS would have the same path (Windows can map to a letter like `I:/` whereas for macOS it's under `/Volumes`) but this part I'm not certain

**Comment by dchiu911** [comment]
Hi @kevinushey one possible solution for our usage is to use `.Rprofile` instead of `.Renviron`:
```r
source("renv/activate.R")
if (.Platform$OS.type == "windows") {
  Sys.setenv(RENV_PATHS_LOCAL = "i:/data repo")
} else {
  Sys.setenv(RENV_PATHS_LOCAL = "/Volumes/.../data repo")
}
```
I think this issue can be closed and revisited if something changes.

**Comment by dchiu911** [comment]
Hi @kevinushey is it possible to supply a vector of multiple locations to `RENV_PATHS_LOCAL`?, e.g.
```
RENV_PATHS_LOCAL="i:/data repo:/Volumes/.../data repo"
```

**Comment by dchiu911** [comment]
Furthermore, since "RemoteUrl" uses absolute paths, everytime a different collaborator updates a local pkg it would change the "RemoteUrl" field in addition to "Version" and "Hash". Is this the intended result?
```
"RemoteUrl": "/Users/person1/Packages/pkgName_0.1.0.tar.gz"
``` 
```
"RemoteUrl": "/Users/person2/Packages/pkgName_0.1.1.tar.gz"
```
If the collaborator is using a different operating system (e.g. Windows) we'll get an even more different "RemoteURL" each time (e.g. "C:/Users/person2/.../).

**Comment by kevinushey** [comment]
Sorry for the late response. Can you elaborate / provide a reproducible example to demonstrate how the package is being installed, and how that RemoteUrl is being added?

**Comment by kevinushey** [comment]
I've also now made a change in the development version of `renv` allowing multiple "local" paths to be set in `RENV_PATHS_LOCAL`, using `;` as a separator.

**Comment by dchiu911** [comment]
Our team's workflow:

1. Copy updated package source tarballs from network drive `i:/data repo/pkgName_0.1.1.tar.gz` to `RENV_PATHS_LOCAL=~/Packages`
2. `renv::install("~/Packages/pkgName_0.1.1.tar.gz")`
3. `renv::snapshot()`
4. At this point, the lock file entry for this package gets updated fields for Version, RemoteUrl, and Hash. And because I set the local path to be "~/Packages", depending on who makes the snapshot, on a mac for e.g. we'd get a field that looks like 
```
"RemoteUrl": "/Users/my_name/Packages/pkgName_0.1.1.tar.gz",
```
where `my_name` would be the person who called snapshot. If we used relative paths such that 
```
"RemoteUrl": "~/Packages/pkgName_0.1.1.tar.gz",
```
then the field would only change for the version number of the package, although I do think using absolute path was the safer intention?

Using multiple local paths would definitely help our team when running from different platforms, I'll try it out thanks!
